### DISCUSSION TITLE
Gene ID from Gene Ontology

### DISCUSSION THREAD

**Discussion by sofiaquintanaesparza** [geneontology/helpdesk#487] [General]
Hi! I'm learning to use R and I was asked to do a task using Gene IDs from Gene Ontology for Homo sapiens. When I download the file from Gene Ontology and open it as a txt, the Gene ID information does not appear. Where can I get it?

**Comment by suzialeksander** [comment]
Hi @sofiaquintanaesparza, can you provide an example of the information you are looking for? In the GO annotation files (GAF 2.2), you can find the gene ID in the first two columns, where we split the database and the database ID. For example the ID `UniProtKB:A0A024RBG1` shows up in a GAF 2.2 in the first two columns as: 

>  UniProtKB	A0A024RBG1 

GO does not make identifiers for proteins/genes/etc., instead we use the identifier provided by the contributing group. This is usually a UniProtKB or MOD-specific (model organism database, like SGD) identifier.

**Reply by sofiaquintanaesparza** [reply]
Yes, you're right, I was confused. The task is about genetic regulation networks using Bingo, the problem is that when applying the command:
universe <- rownames(counts.totals)
genes <- genes
showMethods(genes)
universe
params <- GSEAGOHyperGParams(name="GSEA Parameters",
                             geneSetCollection=gsc,
                             geneIds = genes,
                             universeGeneIds = universe,
                             ontology = "BP",
                             pvalueCutoff = 0.05,
                             conditional = FALSE,
                             testDirection="over")
params
Over <- hyperGTest(params)
Over
It gives me an error called:
> Over <- hyperGTest(params)
Error in getUniverseHelper(probes, datPkg, entrezIds) :
  After filtering, there are no valid IDs that can be used as the Gene universe.
  Check input values ​​to confirm they are the same type as the central ID used by your annotation package.
  For chip packages, this will still mean the central GENE identifier used by the package (NOT the probe IDs).

By reviewing the previous codes, the downloaded files and comparing it with other colleagues, we realized that the GAF file I have is different. In one part of the code, you should get the following:
![WhatsApp Image 2024-05-20 at 15 44 22](https://github.com/geneontology/helpdesk/assets/86010597/832faa10-405c-47b3-82a7-9851285c6fae)
But I get this:
![1](https://github.com/geneontology/helpdesk/assets/86010597/0f41026a-db34-4a56-a27c-f546457f8e24)
So we deduce that the problem is the GAF file that we are using (the commands were given to us by the assistant, so I don't think the error lies there).
I hope I have explained myself well, since I am just learning to use R and I don't understand many things :c

**Reply by sofiaquintanaesparza** [reply]
I took the wrong photo, my file gives me this:
![1](https://github.com/geneontology/helpdesk/assets/86010597/27b4fa87-e3b6-41b2-83ba-065e43ef2afd)

**Comment by suzialeksander** [comment]
Hi @sofiaquintanaesparza,
We are not related to BiNGO, they use our date but we have no involvement. According to their release notes they are using [GAF 2.2](https://geneontology.org/docs/go-annotation-file-gaf-format-2.2/), so the screenshots you are providing do not appear to be the GAF but instead columns you've pulled from the file.  Can you provide a few lines from the raw GAF you're using, and if possible the URL where the GAF is being retrieved? They should look like this:

```
UniProtKB	A0MZ66	SHTN1	enables	GO:0005515	PMID:25416956	IPI	UniProtKB:P05413	F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20240406	IntAct		
UniProtKB	A0MZ66	SHTN1	enables	GO:0005515	PMID:25416956	IPI	UniProtKB:P62508	F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20240406	IntAct		
UniProtKB	A0MZ66	SHTN1	enables	GO:0005515	PMID:25416956	IPI	UniProtKB:Q6PJR7	F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20240406	IntAct		
UniProtKB	A0MZ66	SHTN1	enables	GO:0005515	PMID:25416956	IPI	UniProtKB:Q9Y5B8	F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20240406	IntAct		
UniProtKB	A0MZ66	SHTN1	enables	GO:0019894	GO_REF:0000107	IEA	UniProtKB:Q8K2Q9|ensembl:ENSMUSP00000041378	F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20240129	Ensembl		
UniProtKB	A0MZ66	SHTN1	enables	GO:0045296	PMID:25468996	HDA		F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20180328	BHF-UCL		
UniProtKB	A0MZ66	SHTN1	enables	GO:0051015	GO_REF:0000024	ISS	UniProtKB:A0MZ67	F	Shootin-1	SHTN1|KIAA1598	protein	taxon:9606	20150702	UniProt		

```

**Reply by sofiaquintanaesparza** [reply]
Does it work like this?
![2](https://github.com/geneontology/helpdesk/assets/86010597/0b683696-7be8-43e5-b092-c02c2e20bbb5)
Download it from here
https://current.geneontology.org/products/pages/downloads.html

**Comment by suzialeksander** [comment]
Hi @sofiaquintanaesparza, assuming that the columns V9-V10 are just split for the display, that is the correct file and downloaded from the correct location. The gene ID is in column 2, which you have labelled V2, and technically the ID should have the DB prefix in col 1 (V1) so you might need to cat those and add a `:` separator. See the [GAF specifications](https://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#db-object-id-column-2) for more info.
